test/functional/kaui/admin_allowed_users_controller_test.rb

```diff
@@ -75,24 +82,35 @@ class AdminAllowedUsersControllerTest < Kaui::FunctionalTestHelper
       assert_equal 'User was successfully configured', flash[:notice]
       assert_response :redirect
 
-      au = assigns(:allowed_user)
+      # validate redirect path
+      id = get_allowed_id @response.body
+      assert_equal "/kaui/admin_allowed_users/#{id}", URI(@response.get_header('Location')).path
+
+
       parameters = {
-          :id => au.id,
+          :id => id,
           :allowed_user => { :description => 'An post-apocalyptic super hero' },
           :roles => 'one,two'
       }
 
       put :update, parameters
-
-      allowed_user = assigns(:allowed_user)
-      assert_equal parameters[:allowed_user][:description], allowed_user.description
       assert_equal 'User was successfully updated', flash[:notice]
       assert_response :redirect
+      # validate redirect path
+      assert_equal "/kaui/admin_allowed_users/#{id}", URI(@response.get_header('Location')).path
+
+      # get the user to verify that the data was actually updated
+      get :edit, :id => id
+      description = get_allowed_description @response.body
```

Comment:
Probably just a matter of opinion... 😼 but in such cases of variable assignment, I find it easier to read the code if parens are explicit for methods invocation on the right hand side (i.e. `description = get_allowed_description(@response.body)`).